nixos/modules/services/security/crowdsec.nix

```diff
@@ -0,0 +1,948 @@
+        systemPackages = [ cscli ];
+      };
+
+      systemd.packages = [ cfg.package ];
+
+      systemd.timers.crowdsec-update-hub = lib.mkIf (cfg.autoUpdateService) {
+        description = "Update the crowdsec hub index";
+        wantedBy = [ "timers.target" ];
+        timerConfig = {
+          OnCalendar = "daily";
+          RandomizedDelaySec = 300;
+          Persistent = "yes";
+          Unit = "crowdsec-update-hub.service";
+        };
+      };
+      systemd.services = {
+        crowdsec-update-hub = lib.mkIf (cfg.autoUpdateService) {
+          description = "Update the crowdsec hub index";
+          serviceConfig = {
+            Type = "oneshot";
+            User = cfg.user;
+            Group = cfg.group;
+            LimitNOFILE = 65536;
+            NoNewPrivileges = true;
+            LockPersonality = true;
+            RemoveIPC = true;
+            ReadWritePaths = [
+              rootDir
+              confDir
+            ];
+            ProtectSystem = "strict";
+            PrivateUsers = true;
+            ProtectHome = true;
+            PrivateTmp = true;
+            PrivateDevices = true;
+            ProtectHostname = true;
+            UMask = "0077";
+            ProtectKernelTunables = true;
+            ProtectKernelModules = true;
+            ProtectControlGroups = true;
+            ProtectProc = "invisible";
+            SystemCallFilter = [
+              " " # This is needed to clear the SystemCallFilter existing definitions
+              "~@reboot"
+              "~@swap"
+              "~@obsolete"
+              "~@mount"
+              "~@module"
+              "~@debug"
+              "~@cpu-emulation"
+              "~@clock"
+              "~@raw-io"
+              "~@privileged"
+              "~@resources"
+            ];
+            CapabilityBoundingSet = [
+              " " # Reset all capabilities to an empty set
+            ];
+            RestrictAddressFamilies = [
+              " " # This is needed to clear the RestrictAddressFamilies existing definitions
+              "none" # Remove all addresses families
+              "AF_UNIX"
+              "AF_INET"
+              "AF_INET6"
+            ];
+            DevicePolicy = "closed";
+            ProtectKernelLogs = true;
+            SystemCallArchitectures = "native";
+            RestrictNamespaces = true;
+            RestrictRealtime = true;
+            RestrictSUIDSGID = true;
+            ExecStart = "${lib.getExe cscli} --error hub update";
+            ExecStartPost = "systemctl reload crowdsec.service";
+          };
+        };
+
+        crowdsec = {
+          description = "CrowdSec agent";
+          wantedBy = [ "multi-user.target" ];
+          after = [ "network-online.target" ];
+          wants = [ "network-online.target" ];
+          path = lib.mkForce [ ];
+          environment = {
+            LC_ALL = "C";
+            LANG = "C";
+          };
+          serviceConfig = {
```

Comment:
@acuteaangle I'm a bit unsure if I'm understanding the `DynamicUser` thingy correctly:
So I just need to apply `DynamicUser = true`, right?
Should I give the `crowdsec-update-hub` also the `DynamicUser = true` option? I don't see why I sholdn't.